src/gui/accountsettings.cpp

```diff
@@ -62,6 +62,27 @@
 
 namespace {
 constexpr auto propertyFolderInfo = "folderInfo";
+
+bool showEnableE2eeWithVirtualFilesWarningDialog()
+{
+    QMessageBox e2eeWithVirtualFilesWarningMsgBox(QMessageBox::Warning,
+        "End-to-End Encryption with Virtual Files",
+        "You seem to have the Virtual Files feature enabled on this folder. Virtual"
+        " Files and End-to-End Encryption work not well together at the moment. You"
+        " can allow End-to-End Encryption on this folder, but please keep in mind that"
+        " you need to mark the encrypted folder with \"Make always available locally\" before"
+        " creating or modifying files in it.");
```

Comment:
I'm surprised about that strong statement. Isn't it *only* the implicit hydration which fails? Everything else should work fine AFAIK (even adding files in an otherwise unavailable E2EE folder).

Reply:
Yes, I expected too that it would work, but I tried to create files in a e2ee folder that is not marked as always available, and it failed with an assert. These were the captured logs in case you are interessted:

```
2021-04-22 12:26:10:405 [ info nextcloud.sync.credentials.webflow ]:	request finished
2021-04-22 12:26:10:405 [ info nextcloud.sync.networkjob.put ]:	PUT of "http://localhost:8080/remote.php/dav/files/admin/myencfolder/570887b6bfd74bd88febd1a07b44a782" FINISHED WITH STATUS "OK" QVariant(int, 201) QVariant(QString, "Created")
2021-04-22 12:26:10:405 [ critical nextcloud.sync.csync.vio_local ]:	CreateFileW failed on "\\\\?\\C:\\Users\\felix\\workspace\\desktop\\build\\myfiles\\myencfolder\\New Text Document.txt"
2021-04-22 12:26:10:405 [ warning nextcloud.sync.filesystem ]:	Could not get modification time for "C:/Users/felix/workspace/desktop/build/myfiles/myencfolder/New Text Document.txt" with csync, using QFileInfo
2021-04-22 12:26:10:405 [ info nextcloud.sync.filesystem ]:	File "C:/Users/felix/workspace/desktop/build/myfiles/myencfolder/New Text Document.txt" has changed: size:  0 <-> 0 , mtime:  1619087167 <-> -3600
2021-04-22 12:26:10:405 [ critical nextcloud.sync.vfs.cfapi.wrapper ]:	Could not CreateFile for longpath: "\\\\?\\C:\\Users\\felix\\workspace\\desktop\\build\\myfiles\\myencfolder\\New Text Document.txt" with error: 2
2021-04-22 12:26:10:405 [ fatal default ]:	ASSERT: "handle" in file ..\src\libsync\vfs\cfapi\cfapiwrapper.cpp, line 576
```

Reply:
I tried it again and now I was able to create e2ee files, even if the folder is not set to make available offline. Maybe the other error came frome something else. Anyway I think it's a good idea to not create e2ee folders when using vfs.

Reply:
Not getting the modtime of a file you created is indeed odd. Seems vfs related but I somehow doubt this is E2EE related. Could be wrong of course, can't beat having a debugger attached. ;-)

Reply:
I'd tone it down in any case. Right now it's going to scare away users thinking it'll eat their puppies.

Reply:
Do you like it now more?